### PR TITLE
Update nix files to build berkeley migration tools

### PR DESCRIPTION
**Pull request by georgeee**
Apply `caqti-upgrade-plus-archive-init-speedup.patch` to sources before building the packages.

Explain how you tested your changes:
* Built `berkeley` package with `nix build .?submodules=1#mainnet-caqti-patched.berkeley_migration`
* Ran incremental migration pass with `mina-berkeley-migration-script incremental -g mainnet.json -b mina_network_block_data -bs 50 -n mainnet -s postgres://mina@localhost:5440/archive_feb19 -t postgres://mina@localhost:5440/migrated -r migration-checkpoint-497736.json`

Checklist:

- [x] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [x] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [x] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? None

### CODE REVIEW

File: nix/ocaml.nix [hunk 1]
```diff
@@ -277,35 +276,6 @@ let
 
       devnet = wrapMina self.devnet-pkg { };
 
-      berkeley-migration-pkg = self.mina-dev.overrideAttrs (s: {
```
Comment (emberian) [inline nix/ocaml.nix:280]: one reason we separated these out in the debs was overlapping package conflicts. but I imagine that's simply not a problem with nix.

the only requirement is then to ~always disambiguate which thing you want packages from? vs adding to user environment.

**Review by emberian — APPROVED** [review]

File: nix/ocaml.nix (hunk 1 above)
Comment (georgeee) [reply nix/ocaml.nix:280]: Yeah, it was in this form before because different `DUNE_PROFILE` was used, but this is not the case anymore.

**Comment by georgeee** [comment]
!ci-build-me

**Review by dkijania — APPROVED** [review]

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**